### PR TITLE
Fixing windows resource not found error

### PR DESCRIPTION
**Pull request by abelet**

### CODE REVIEW

**Comment by abelet** [comment]
The console could not be built on  windows before the modification

File: processors/src/main/java/org/jboss/hal/processor/mbui/MbuiViewProcessor.java [hunk 1]
```diff
@@ -234,7 +233,7 @@ private Document parseXml(TypeElement type, MbuiView mbuiView) {
         String mbuiXml = Strings.isNullOrEmpty(mbuiView.value())
                 ? type.getSimpleName().toString() + ".mbui.xml"
                 : mbuiView.value();
-        String fq = TypeSimplifier.packageNameOf(type).replace('.', '/') + File.separator + mbuiXml;
+        String fq = TypeSimplifier.packageNameOf(type).replace('.', '/') + "/" + mbuiXml;
```
Comment (hpehl) [inline processors/src/main/java/org/jboss/hal/processor/mbui/MbuiViewProcessor.java:236]: Why doesn't it compile on Windows? 
Shouldn't that be rather:
```java
String fq = TypeSimplifier.packageNameOf(type).replace('.', File.separatorChar) + File.separator + mbuiXml;
```

**Comment by abelet** [comment]
Before suggesting a solution I tried the same you guessed, and it gave the followin error:

```java
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0-jboss-2:compile (default-compile) on project hal-console: Fatal error compiling: java.lang.IllegalArgumentException: Invalid relative name: org\jboss\hal\client\configuration\subsystem\u
ndertow\BufferCacheView.mbui.xml -> [Help 1]
```

**Comment by abelet** [comment]
I needed the modification to check a bug (https://issues.redhat.com/browse/HAL-1681) in HAL. 
This bug (HAL-1681) hasn't been assigned to anyone. Do you think is there any hope a fix will be provided?

**Comment by hpehl** [comment]
Thanks for bringing this up @abelet. I can see the problem and I think we should add `.param(RESOLVE_EXPRESSIONS, true)` for all `:read-resource` operations in the runtime section of HAL. But this would only fix the operations which are not recursive. I'll try to fix the HAL part asap.

Regarding the original issue with build failures on windows, I'm going to merge your fix.

**Comment by abelet** [comment]
I reported an issue about the recursive expression resolution. The solution is just one line, so I create a pull request if nothing happens.
In the HAL what do you think something like the following (this way no chance that something won't work in the future when a new readresource operation is added and the modification is very easy):


```java
package org.jboss.hal.dmr;
....

@JsType
public class Operation extends ModelNode {
...
        @JsIgnore
        public Builder(ResourceAddress address, String name) {
            this.address = address;
            this.name = name;
            this.parameter = new ModelNode();
            this.header = new ModelNode();
            this.roles = new HashSet<>();
            if(name.equals(READ_RESOURCE_OPERATION)) {
              parameter.get(RESOLVE_EXPRESSIONS).set(true);
            }
        }
```

**Comment by hpehl** [comment]
I'm not sure if we really should add `RESOLVE_EXPRESSIONS` to each and every `:read-resource` operation. I'd like that to discuss with my peers first.

**Pull request merged**